packages/react-native/React/Base/RCTConvert.m

```diff
@@ -1487,6 +1487,105 @@ + (NSPropertyList)NSPropertyList:(id)json
     integerValue)
 
 #if TARGET_OS_OSX // [macOS
+// Used to convert the JSON from `role` to the corresponding
+// NSAccessibilityRole (NSString*), has slightly different mappings than the
+// old accessibilityRole prop for back compatability. `role` matches ARIA Core
+// AAM spec and takes precedence. See https://www.w3.org/TR/core-aam-1.1/
++ (NSString*) accessibilityRoleFromRole:(NSString*)ariaRole
+{
+  static NSDictionary<NSString *, NSAccessibilityRole> * ariaRoleToNsRole;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    ariaRoleToNsRole = @{
+      @"alert": NSAccessibilityGroupRole,
+      @"alertdialog": NSAccessibilityGroupRole,
+      @"application": NSAccessibilityGroupRole,
+      @"article": NSAccessibilityGroupRole,
+      @"banner": NSAccessibilityGroupRole,
+      @"button": NSAccessibilityButtonRole,
+      @"cell": NSAccessibilityCellRole,
+      @"checkbox": NSAccessibilityCheckBoxRole,
+      @"columnheader": NSAccessibilityCellRole,
+      @"combobox": NSAccessibilityComboBoxRole,
+      @"complementary": NSAccessibilityGroupRole,
+      @"contentinfo": NSAccessibilityGroupRole,
+      @"definition": NSAccessibilityGroupRole,
+      @"dialog": NSAccessibilityGroupRole,
+      @"directory": NSAccessibilityListRole,
+      @"document": NSAccessibilityGroupRole,
+      @"feed": NSAccessibilityGroupRole,
+      @"figure": NSAccessibilityGroupRole,
+      @"form": NSAccessibilityGroupRole,
+      @"grid": NSAccessibilityTableRole,
+      @"gridcell": NSAccessibilityCellRole,
+      @"group": NSAccessibilityGroupRole,
+      @"heading": NSAccessibilityStaticTextRole,
+      @"image": NSAccessibilityImageRole,
+      @"img": NSAccessibilityImageRole,
+      @"link": NSAccessibilityLinkRole,
+      @"list": NSAccessibilityListRole,
+      @"listbox": NSAccessibilityListRole,
+      @"listitem": NSAccessibilityGroupRole,
+      @"log": NSAccessibilityGroupRole,
+      @"main": NSAccessibilityGroupRole,
+      @"marquee": NSAccessibilityGroupRole,
+      @"math": NSAccessibilityGroupRole,
+      @"menu": NSAccessibilityMenuRole,
+      @"menubar": NSAccessibilityMenuBarRole,
+      @"menuitem": NSAccessibilityMenuItemRole,
+      @"menuitemcheckbox": NSAccessibilityMenuItemRole,
+      @"menuitemradio": NSAccessibilityMenuItemRole,
+      @"meter": NSAccessibilityLevelIndicatorRole,
+      @"navigation": NSAccessibilityGroupRole,
+      @"none": NSAccessibilityGroupRole,
+      @"note": NSAccessibilityGroupRole,
+      @"option": NSAccessibilityStaticTextRole,
+      @"presentation": NSAccessibilityGroupRole,
+      @"progressbar": NSAccessibilityProgressIndicatorRole,
+      @"radio": NSAccessibilityRadioButtonRole,
+      @"radiogroup": NSAccessibilityRadioGroupRole,
+      @"region": NSAccessibilityGroupRole,
+      @"row": NSAccessibilityRowRole,
+      // rowgroup not mapped
```

Comment:
Is there like an unknown role or is it better to leave omitted?

Reply:
I debated this too. The ARIA spec is a little vague about what "Not Mapped" means in this instance.  The current implementation is going to fail to find the key in the dictionary and fall back to `NSAccessibilityUnknownRole`.  Arguably, I think we might be better off using the defaultView's accessibilityRole in that situation.  Do you have an opinion?

Reply:
If by `defaultView` you mean the one that argument that shows up in RCTConvert / view manager code, I don't think that's ever used. If you mean like the default accessibilityRole of an NSView, what is that? I thought it might be nil and then VoiceOver would just skip over it?